### ISSUE TITLE
bug: mbsync passwordCommand fails to run

### ISSUE THREAD

**Issue by mmequignon** [nix-community/home-manager#3650]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

my config looks like this
```
accounts.email.email_account.password_command = "${pkgs.pass}/bin/pass email_account | ${pkgs.coreutils}/bin/head -n 1"
```
I now got this in my `.mbyncrc`
```
[...]
PassCmd "'/nix/store/.../bin/pass' 'mail_account' '|' '/nix/store/.../bin/head' '-n' '1'"
[...]
```
Which is wrongly evaluated by mbsync
```
Skipping account mail_account, PassCmd exited with status 1
```
Astroid outputs more helpful logs
```
[19:49:18] [0x00007ff4b1c58b40] [M] [warning] poll script: pass: invalid option -- 'n'
[19:49:18] [0x00007ff4b1c58b40] [M] [warning] poll script: Usage: pass show [--clip[=line-number],-c[line-number]] [--qrcode[=line-number],-q[line-number]] [pass-name]
[19:49:18] [0x00007ff4b1c58b40] [M] [warning] poll script: Skipping account mail_account, PassCmd exited with status 
```

Looks like it is related to https://github.com/nix-community/home-manager/pull/3630

Thanks!

### Maintainer CC

@ncfavier @IWYpPtD49k1OHyy3AqwT9vOBJEaZ4S 

### System information

```markdown
- system: `"x86_64-linux"`
 - host os: `Linux 6.1.7, NixOS, 22.11 (Raccoon), 22.11.1777.cdead16a444`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"nixos-22.11, nixpkgs"`
 - channels(matthieu): `"home-manager, nixpkgs, nixpkgs-unstable"`
 - nixpkgs: `/home/matthieu/.nix-defexpr/channels/nixpkgs`
```

**Comment by mmequignon** [comment]
If I checkout to https://github.com/nix-community/home-manager/commit/d1c7730bb707bf8124d997952f7babd2a281ae68, it works

**Comment by ncfavier** [comment]
I was a bit fast to merge https://github.com/nix-community/home-manager/pull/3630 and did not think properly about the backwards compatibility breakage.

On one hand `passwordCommand` is usually interpreted as an execve-style command and it would be nice to have a consistent interface, but on the other hand there's no reason to disallow setting a shell command if mbsync supports it.

@IWYpPtD49k1OHyy3AqwT9vOBJEaZ4S I'm thinking maybe for mbsync we could treat strings as shell commands (`id`) and *lists* of strings as execve commands (`escapeShellArgs`). The only difficulty is that [`passwordCommand`](https://github.com/nix-community/home-manager/blob/b70362bf9bdefcf1042af3848f7311171b07a7cc/modules/accounts/email.nix#L293-L302) currently has an `apply` which turns strings into lists by splitting on whitespace, which is undesirable so it would have to be moved down into all the other modules that don't support shell commands.

**Comment by mmequignon** [comment]
I'm not really familiar with nix, but I'll gladly help on this :blush:

**Comment by ncfavier** [comment]
FWIW as a workaround you can use a script:

```nix
passwordCommand = [ "${pkgs.writeShellScript "password-command" ''
  pass foo | head -n 1
''}" ];
```

**Comment by mmequignon** [comment]
Thank you!
I'm not on my personal computer right now, I'll give it a try later

**Comment by ncfavier** [comment]
I've reverted the PR for now. https://github.com/nix-community/home-manager/pull/3657

**Comment by rycee** [comment]
Yeah, perhaps best to make it clear in the `passwordCommand` option description that the command is _not_ a shell command but rather a straight execv style command. It is purely accidental if a pipe should happen to work. If one wants to do such things then the correct solution would be to use something like `pkgs.writeScript`.
